zerver/tests/test_message_flags.py

```diff
@@ -1098,17 +1098,6 @@ def test_change_star_public_stream_historical(self) -> None:
             {"messages": orjson.dumps(sent_message_ids).decode(), "op": "add", "flag": "read"},
         )
 
-        # We can't change flags other than "starred" on historical messages:
-        result = self.client_post(
-            "/json/messages/flags",
-            {"messages": orjson.dumps(message_ids).decode(), "op": "add", "flag": "read"},
-        )
-        self.assert_json_error(result, "Invalid message(s)")
-
-        # Trying to change a list of more than one historical message fails
-        result = self.change_star(message_ids * 2)
-        self.assert_json_error(result, "Invalid message(s)")
-
```

Comment:
Should we replace this test with one that passes that toggles an additional flag, say `collapsed`?  I think this commit makes that now possible, which is actually a slightly useful feature.

Reply:
fixed in upcoming push

Reply:
(I added the test for "collapsed".)

We have this:

~~~
3091     # Certain flags are used only for internal accounting within the
3092     # Zulip backend, and don't make sense to expose to the API.
3093     NON_API_FLAGS = {"is_private", "active_mobile_push_notification"}
3094     # Certain additional flags are just set once when the UserMessage
3095     # row is created.
3096     NON_EDITABLE_FLAGS = {
3097         # These flags are bookkeeping and don't make sense to edit.
3098         "has_alert_word",
3099         "mentioned",
3100         "wildcard_mentioned",
3101         "historical",
3102         # Unused flags can't be edited.
3103         "force_expand",
3104         "force_collapse",
3105         "summarize_in_home",
3106         "summarize_in_stream",
3107     }

Reply:
And:

~~~ py
6359     valid_flags = [item for item in UserMessage.flags if item not in UserMessage.NON_API_FLAGS]
6360     if flag not in valid_flags:
6361         raise JsonableError(_("Invalid flag: '{}'").format(flag))
6362     if flag in UserMessage.NON_EDITABLE_FLAGS:
6363         raise JsonableError(_("Flag not editable: '{}'").format(flag))

Reply:
I guess my question was about the tests; I guess the fact we require 100% test coverage for `actions.py` means this should be OK; and I did a quick verification.